packages/node/src/stellar/api.service.stellar.ts

```diff
@@ -52,6 +53,10 @@ export class StellarApiService extends ApiService<
   ): Promise<StellarApiService> {
     let network: StellarProjectNetworkConfig;
 
+    const needSorobanTxMeta = !!project.dataSources.find(
```

Comment:
This needs to take into account project upgrades and dynamic datasoruces.

Does the user requesting this feature also need to access the transactions from soroban events?